src/app/wallets/ln-send-payment.ts

```diff
@@ -483,7 +493,16 @@ const executePaymentViaLn = async ({
         })
     if (payResult instanceof LnPaymentPendingError) return PaymentSendStatus.Pending
```

Comment:
what is the flow is the payment return an error here? (not the case of a pending error but the case the payment simply failed)

Reply:
The conditional would fail and this return ignored. All other errors not explicitly handled would then be caught at [`Ln 508`](https://github.com/GaloyMoney/galoy/blob/persist-ln-payment-in-medici/src/app/wallets/ln-send-payment.ts#L508).

Reply:
are you sure the errors from the `lightning` package would be caught correctly?

lightning doesn't return an `Error` instance, but an array (the custom way to doing error reporting was not intuitive to me at first)

therefore I would not think `if (payResult instanceof Error) {` would be trigger with a payment error because payResult would be an `Array`, not an `Error`

Reply:
It looks like we handle the errors we know about that get thrown from lightning (see [here](https://github.com/GaloyMoney/galoy/blob/041fd608e022d32fcf512463766f51ea3f967f9f/src/services/lnd/index.ts#L169)) and then for the ones we don't know about we wrap the array in an `UnknownLightningServiceError` object and bubble it up to be handled by calling code.

So in this case as I understand now, any error that isn't an `isInvoiceAlreadyPaidError` error would get put into that wrapper and would be caught by the `if (payResult instanceof Error) {` line